File: Medium/2026-04-16-3488-Closest-Equal-Element-Queries/solution.js
var closestEqualElementQueries = function(nums, queries) {
    const n = nums.length;
    const indexMap = new Map();
    const answer = new Array(queries.length).fill(-1);
    
    for (let i = 0; i < n; i++) {
        if (!indexMap.has(nums[i])) {
            indexMap.set(nums[i], []);
        }
        indexMap.get(nums[i]).push(i);
    }
    
    for (let i = 0; i < queries.length; i++) {
        const queryIndex = queries[i];
        const value = nums[queryIndex];
        const indices = indexMap.get(value);
        
        if (indices.length <= 1) continue;
        
        let minDistance = Infinity;
        for (const index of indices) {
            if (index !== queryIndex) {
                const distance = Math.abs(index - queryIndex);
                const circularDistance = n - distance;
                minDistance = Math.min(minDistance, Math.min(distance, circularDistance));
            }
        }
        
        answer[i] = minDistance === Infinity ? -1 : minDistance;
    }
    
    return answer;
};

File: Medium/2026-04-16-3488-Closest-Equal-Element-Queries/solution.java
import java.util.*;

public class Solution {
    public int[] closestEqualElementQueries(int[] nums, int[] queries) {
        Map<Integer, List<Integer>> indexMap = new HashMap<>();
        int n = nums.length;
        
        // Populate the index map with positions of each number
        for (int i = 0; i < n; i++) {
            indexMap.computeIfAbsent(nums[i], k -> new ArrayList<>()).add(i);
        }
        
        int[] answer = new int[queries.length];
        
        for (int i = 0; i < queries.length; i++) {
            int queryIndex = queries[i];
            int value = nums[queryIndex];
            List<Integer> positions = indexMap.get(value);
            
            if (positions.size() <= 1) {
                answer[i] = -1;
                continue;
            }
            
            int minDistance = Integer.MAX_VALUE;
            for (int pos : positions) {
                if (pos != queryIndex) {
                    int distance = Math.abs(pos - queryIndex);
                    distance = Math.min(distance, n - distance); // Circular distance
                    minDistance = Math.min(minDistance, distance);
                }
            }
            answer[i] = minDistance;
        }
        
        return answer;
    }
}